hub/core/chunk_engine.py

```diff
@@ -543,6 +546,25 @@ def append(self, sample: SampleValue):
         """Formats a single `sample` (compresseses/decompresses if applicable) and feeds it into `_append_bytes`."""
         self.extend([sample])
 
+    def clear(self):
+        """Deletes all samples."""
+        self.cache.check_readonly()
+
+        n_samples = self.num_samples
```

Comment:
Is this scenario handled: one client is trying to delete and another is appending samples to the same tensor?

Reply:
@jraman that can not be handled from the scope of this PR.